packages/fetch-router/src/lib/route-map.ts

```diff
@@ -4,39 +4,39 @@ import type { HrefBuilderArgs, Join, RouteMatch } from '@remix-run/route-pattern
 import type { RequestMethod } from './request-methods.ts'
 import type { Simplify } from './type-utils.ts'
 
-export interface RouteMap<T extends string = string> {
-  [K: string]: Route<RequestMethod | 'ANY', T> | RouteMap<T>
+export interface RouteMap<name extends string = string> {
+  [name: string]: Route<RequestMethod | 'ANY', name> | RouteMap<name>
```

Comment:
Shouldn't this line be something like:

```ts
  [key: string]: Route<RequestMethod | 'ANY', name> | RouteMap<name>
```

using `key` instead of `name` for the index variable?

Naming the index variable the same as the generic is a little confusing.